test/node/double_tests.js

```diff
@@ -35,4 +36,74 @@ describe('Double', function () {
       });
     }
   });
+
+  describe('specialValues', () => {
+    function twiceSerialized(value) {
+      let serializedDouble = BSON.serialize({ d: value });
+      let deserializedDouble = BSON.deserialize(serializedDouble, { promoteValues: true });
+      let newVal = deserializedDouble.d;
+      return newVal;
+    }
+
+    it('inf', () => {
+      let value = Infinity;
+      let newVal = twiceSerialized(value);
+      expect(value).to.equal(newVal);
+    });
+
+    it('-inf', () => {
+      let value = -Infinity;
+      let newVal = twiceSerialized(value);
+      expect(value).to.equal(newVal);
+    });
+
+    it('NaN', () => {
+      let value = NaN;
+      let newVal = twiceSerialized(value);
+      expect(Number.isNaN(newVal)).to.equal(true);
+    });
+
+    it('NaN with payload', function () {
```

Comment:
There's a lot of hard to follow logic in this test that doesn't make it clear what the assertions are really trying to accomplish, please clarify the description and the setup, possibly splitting things up into more than one test case as appropriate.

Reply:
That's right, I've refactored and added more descriptive variable names to make it more readable